modules/code-deployment/src/main/java/org/apache/ignite/internal/deployunit/UnitMetaSerializer.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.deployunit;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import org.apache.ignite.deployment.version.Version;
+
+/**
+ * Serializer for {@link UnitMeta}.
+ */
+public class UnitMetaSerializer {
+    private static final String SEPARATOR = ";";
+    private static final String LIST_SEPARATOR = ":";
+
+    /**
+     * Constructor.
+     */
+    private UnitMetaSerializer() {
+
+    }
+
+
+    /**
+     * Serialize unit meta.
+     *
+     * @param meta Unit meta.
+     * @return Serialized unit meta.
+     */
+    public static byte[] serialize(UnitMeta meta) {
+        StringBuilder sb = new StringBuilder();
+
+        sb.append(meta.getId()).append(SEPARATOR)
+                .append(meta.getVersion().render()).append(SEPARATOR)
+                .append(meta.getUnitName()).append(SEPARATOR);
```

Comment:
What if `id`, `name` or `version` contains separator or list separator?

Reply:
Right, but id and name are just strings, no?

Reply:
Ok, but keep in mind, that in
```
for (String id : meta.getConsistentIdLocation()) {
    sb.append(id).append(LIST_SEPARATOR);
}
```
you can also encounter node names with colons. Escaping your strings is the thing that you should always keep in mind during such concatenations

Reply:
We can just use base64 for serialization

Reply:
Does base64 include colon? I think it does. So we should use base63 instead